src/common/asb/Asb.c

```diff
@@ -3654,9 +3436,19 @@ static int RemediateEnsureSmbWithSambaIsDisabled(char* value, void* log)
 
 static int RemediateEnsureUsersDotFilesArentGroupOrWorldWritable(char* value, void* log)
 {
-    unsigned int modes[] = {600, 644, 664, 700, 744};
-    UNUSED(value);
-    return SetUsersRestrictedDotFiles(modes, ARRAY_SIZE(modes), 744, log);
+    int* modes = NULL;
+    int numberOfModes = 0;
+    int status = 0;
+
+    InitEnsureUsersDotFilesArentGroupOrWorldWritable(value);
+
+    if (0 == (status = ConvertStringToIntegers(g_desiredEnsureUsersDotFilesArentGroupOrWorldWritable, ',', &modes, &numberOfModes, log)))
+    {
+        status = SetUsersRestrictedDotFiles((unsigned int*)modes, (unsigned int)numberOfModes, modes[(numberOfModes > 1) ? (numberOfModes - 1) : 0], log);
```

Comment:
Is `numberOfModes` guaranteed to be non-zero here?

Reply:
Very good question. ConvertStringToIntegers will always return at least one 0 value for any non NULL string passed in, except failing with EINVAL when there is no string. Stay tuned.